src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingServiceRemotingClient.cs

```diff
@@ -0,0 +1,177 @@
+        /// <param name="requestBody">Message body.</param>
+        public void SendOneWay(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody)
+        {
+            SendAndTrackRequestAsync(messageHeaders, requestBody, () =>
+            {
+                this.InnerClient.SendOneWay(messageHeaders, requestBody);
+                return Task.FromResult<byte[]>(null);
+            }).Forget();
+        }
+
+        private async Task<byte[]> SendAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, byte[] requestBody, Func<Task<byte[]>> doSendRequest)
+        {
+            DependencyTelemetry dt = new DependencyTelemetry();
+            dt.Name = ServiceRemotingLoggingStrings.OutboundRequestActivityName;
+
+            // Determine if there is currently an activity. If there is, initializes the dependency telemetry with the activity information,
+            // which helps set the right context for the dependency telemetry
+            Activity currentActivity = Activity.Current;
+            if (currentActivity != null)
+            {
+                dt.Id = currentActivity.Id;
+                dt.Context.Operation.Id = currentActivity.RootId;
+                dt.Context.Operation.ParentId = currentActivity.ParentId;
+            }
+
+            this.fabricTelemetryInitializer.Initialize(dt);
+
+            // Call StartOperation, this will create a new activity with the current activity being the parent. This
+            // new activity also inherits most of the properties/attributes carried by the DependencyTelemetry object.
+            var operation = telemetryClient.StartOperation<DependencyTelemetry>(dt);
+
+            bool success = true;
+            try
+            {
+                // After the operation, the activity stack, if any, will have been changed. We need
+                // to update what our current activity is.
+                currentActivity = Activity.Current;
+                if (currentActivity != null)
+                {
+                    messageHeaders.AddHeader(ServiceRemotingLoggingStrings.RequestIdHeaderName, currentActivity.Id);
+
+                    // We expect the baggage to not be there at all or just contain a few small items
+                    if (currentActivity.Baggage.Any())
+                    {
+                        using (var ms = new MemoryStream())
+                        {
+                            var dictionaryWriter = XmlDictionaryWriter.CreateBinaryWriter(ms);
+                            this.baggageSerializer.Value.WriteObject(dictionaryWriter, currentActivity.Baggage);
+                            dictionaryWriter.Flush();
+                            messageHeaders.AddHeader(ServiceRemotingLoggingStrings.CorrelationContextHeaderName, ms.GetBuffer());
+                        }
+                    }
+                }
+
+                byte[] result = await doSendRequest();
```

Comment:
ConfigureAwait(false) should always be used in  a library

https://medium.com/bynder-tech/c-why-you-should-use-configureawait-false-in-your-library-code-d7837dce3d7f